x-pack/plugins/apm/public/components/shared/Links/url_helpers.ts

```diff
@@ -12,7 +12,9 @@ export function toQuery(search?: string): APMQueryParamsRaw {
 }
 
 export function fromQuery(query: StringMap<any>) {
-  return qs.stringify(query);
+  return qs.stringify(query, undefined, undefined, {
+    encodeURIComponent: (value: string) => value
```

Comment:
curious, I assume the `toQuery` function expects encoded characters, and will attempt to decode it. if so, will this not break things in weird ways? (I guess a unit test for toQuery will expose whether this is the case)

Reply:
What do you think about this existing test:

https://github.com/elastic/kibana/blob/ee02788f3225579280974e03d478396d4a21b2ed/x-pack/plugins/apm/public/components/shared/Links/url_helpers.test.tsx#L16-L23

In the test `toQuery` correctly decodes `john%20doe` to `john doe`. Should I add an additional test for decoding?

Reply:
I think the change is that it is no longer encoded though. As an example, suppose the user types in "foo:bar" as a search. that will be passed to toQuery as "foo:bar", where before it was "foo%3Abar". Now, "foo:bar" will still decode to "foo:bar". But what happens if the search that the user entered is "foo:%20"? It will be decoded as "foo: ", as far as I can tell.

Reply:
God point. I'll add a test for this and see how it behaves.

Reply:
@dgieselaar You were right. I updated the behaviour so it encodes everything but `:` (colon) and wrote a test for it.

Reply:
👍 I guess there is still then the edge case of `%3A`, but I'm not sure how it should be solved either (other than removing Angular from the equation). URLs are hard!